src/components/Radio/index.tsx

```diff
@@ -0,0 +1,108 @@
+import type { ChangeEvent, HTMLAttributes, ReactElement } from 'react'
+import styled from '@emotion/styled'
+import type { StyledProps } from '@types'
+
+export interface RadioProps extends HTMLAttributes<HTMLDivElement> {
+  formName: string
+  items: { code: string; name: string }[]
+  direction: 'horizontal' | 'vertical'
+  onChange(e: ChangeEvent<HTMLInputElement>): void
+}
+
+type StyledFormProps = StyledProps<RadioProps, 'direction'>
+
+export const Radio = ({
+  formName,
+  onChange,
+  items,
+  direction = 'horizontal',
+  ...props
+}: RadioProps): ReactElement => {
+  const handleRadiobutton = (e: ChangeEvent<HTMLInputElement>): void => {
+    onChange(e)
+  }
+
+  const radioList = items?.map(({ code, name }) => (
+    <StyledInputWrapper key={code} className={`${direction}`}>
+      {name}
+      <StyledInput
+        id={code}
+        name={formName}
+        type="radio"
+        value={code}
+        onChange={handleRadiobutton}
+      />
+      <StyledCheckMark />
+    </StyledInputWrapper>
+  ))
+
+  return (
+    <div {...props}>
+      <StyledForm direction={direction}>{radioList}</StyledForm>
+    </div>
```

Comment:
위에서 div가 아닌 form 타입을 extends하면 form에 props 전달이 가능할 것 같아용!

```suggestion
  return (
      <StyledForm direction={direction} {...props}>{radioList}</StyledForm>
```

Reply:
요거 내부에서 form extends 받게되면 onChange 네이밍으로 지정 못해주는 이슈가 있어서 div로 받았는데 form으로 받게되면 & 로 받아야할 것 같네요!

Reply:
네, 수림넴 말씀처럼 onchange 네이밍으로 지정 못해줘가지구 다시 바꾸어 놓았는데, form으로 받는 것이 나을까요??

Reply:
흠 제 생각엔 form이 중점인 컴포넌트인 만큼 사용자가 form extends를 기대할거라고 생각이 들어요 
그래서 변경하는게 더 낫다고 생각합니다!